### PR TITLE
Bug 2054351:Fix restart process  that was holding  lock on asynchronous call, avoid multiple socket error 

### PR DESCRIPTION
**Pull request by aneeshkp**
Signed-off-by: Aneesh Puttur <aneeshputtur@gmail.com>
Cleaning termination with an exit in defer function and not holding the lock for asynchronous calls like cmd start and sig term.
This was causing the listener at the sidecar to see multiple socket connections (2 per termination)

### CODE REVIEW

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -363,43 +384,35 @@ func cmdRun(p *ptpProcess, stdoutToSocket bool) {
...
-
 		err = p.cmd.Wait()
 		if err != nil {
 			glog.Errorf("cmdRun() error waiting for %s: %v", p.name, err)
 		}
 
 		time.Sleep(connectionRetryInterval) // Delay to prevent flooding restarts if startup fails
-		p.execMutex.Lock()                  // Don't restart after termination
+		// Don't restart after termination
 		if p.stopped {
```
Comment (josephdrichard) [inline pkg/daemon/daemon.go:402]: p.Stopped()

File: pkg/daemon/daemon.go [hunk 2]
```diff
@@ -312,6 +325,14 @@ func ptp4lCreateCmd(nodeProfile *ptpv1.PtpProfile, confFilePath string) *exec.Cm
 func cmdRun(p *ptpProcess, stdoutToSocket bool) {
 	var c net.Conn
 	done := make(chan struct{}) // Done setting up logging.  Go ahead and wait for process
+	defer func() {
```
Comment (josephdrichard) [inline pkg/daemon/daemon.go:328]: Is everything outside of this defer func just style changes, or necessary for this fix?
If it's not part of this fix, can you put it in a separate commit?

**Review by josephdrichard — COMMENTED** [review]
Is there an open bug for this?

File: pkg/daemon/daemon.go (hunk 2 above)
Comment (aneeshkp) [reply pkg/daemon/daemon.go:328]: This is part of the restart process and I want to backport this once it is thoroughly tested by QE.
So it is good to keep it in one PR to track the fix or for any issues to reverse the changes.

Comment (josephdrichard) [reply pkg/daemon/daemon.go:328]: Can you split it into two separate commits within the same PR?

**Comment by josephdrichard** [comment]
Commit message seems to have been incorrectly changed.

File: pkg/daemon/daemon.go (hunk 2 above)
Comment (aneeshkp) [reply pkg/daemon/daemon.go:328]: I don't get it , what two PR?

**Comment by aneeshkp** [comment]
> > Is there an open bug for this?
> 
> @josephdrichard , Haven't created a bug for this. I spoke to Ken and told him we will create a bug if we backport only, that is after successful testing. So as of now, no plans to do any backport PR immediately.

created bug and updated

**Comment by aneeshkp** [comment]
> Commit message seems to have been incorrectly changed.

amended the commit message , Thanks
